pkg/ignite/ignite.go

```diff
@@ -1,5 +1,53 @@
 package ignite
 
-const (
-	execName = "ignite"
+import (
+	"fmt"
+	"regexp"
+	"strconv"
+
+	log "github.com/sirupsen/logrus"
+	"github.com/weaveworks/footloose/pkg/exec"
+)
+
+const execName = "ignite"
+
+var (
+	versionRegex = regexp.MustCompile(`^v(\d+)\.(\d+)\.(\d+)`)
+	minVersion   = []int{0, 5, 2} // Major, minor, patch
 )
+
+func CheckVersion() {
```

Comment:
can you use a library (e.g. https://github.com/blang/semver) instead; that'll make this code more readable and easy-to-use.

some cases that should be supported:
 - v0.5.1 should fail
 - v0.5.2 and v0.5.3+ should succeed
 - v0.6.0 should succeed
 - a dirty build should just throw a warning that "you're using a dirty build of ignite; continuing but your mileage may vary" or similar

Reply:
Sure thing, didn't know if we need to pull in a new dependency for version checking but I can refactor.